mapit-backend/src/Controller/TagController.php

```diff
@@ -31,6 +35,49 @@ public function index(TagRepository $tagRepository, TagSerializer $tagSerializer
      */
     public function create(Request $request, TagRepository $tagRepository, TagSerializer $tagSerializer): JsonResponse {
         $tag = $tagSerializer->deserialize($request->getContent());
+        $tagRepository->save($tag); 
+
+        return new JsonResponse(
+            $tagSerializer->serialize($tag),
+            JsonResponse::HTTP_OK,
+            [],
+            true
+        );
+    }
+
+    /**
+     * @Route("/tag/add"), methods={"POST"}
+     */
+    public function add(Request $request, TagRepository $tagRepository, PlaceRepository $placeRepository, TagSerializer $tagSerializer, PlaceSerializer $placeSerializer): JsonResponse {
+        $postData = $tagSerializer->deserialize($request->getContent());
+
+        $tagExists = $tagRepository->findOneBy([
+            'name' => $postData->getName()
+        ]);
+
+        if(!($tagExists)) {
+            $tag = new Tag();
+            $tag->setName($postData->getName());
+            $tagExists = $tag;
+        } 
+        
+        $tag = $tagExists;
+
```

Comment:
you coult name the variable in line 54 $tag directly.

Reply:
Fair point – I will do the same with $placeExists.